emails/models.py

```diff
@@ -107,7 +107,26 @@ def has_unlimited(self):
                 return True
         return False
 
-
+    def add_subdomain(self, subdomain):
+        # checking for premium user happens before this funciton is called
```

Comment:
Should we call the check for premium subscription here too? Just to be safe?

Reply:
I took some time to think about this. I didn't add the check for premium subscription here because (1) that was not being done to `make_domain_address` and (2) because `make_relay_address` does NOT check business logic (premium user, else check there are less than 5 aliases already created).

But, I do agree adding an extra check here will be nice so we don't accidentally allow user to be able to add a subdomain. Wondering if we should go ahead and add the check to the `make_relay_address` and `make_domain_address` and remove the check from the emails/views.py?

Reply:
Completed by 96bc252, bb2ae8c, and f666157